TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -16,38 +16,72 @@ namespace TypeCobol.Compiler.Nodes {
         }
-        private int WhereShouldIAdd(System.Type section) {
-            if (Tools.Reflection.IsTypeOf(section, typeof(FileSection))) return 0;
-            int ifile = -2;
-            int iglobal = -2;
-            int iworking = -2;
-            int ilocal = -2;
-            int ilinkage = -2;
-            int c = 0;
-            foreach(var child in this.Children()) {
-                if (Tools.Reflection.IsTypeOf(child.GetType(), typeof(FileSection))) ifile = c;
-                else
-                if (Tools.Reflection.IsTypeOf(child.GetType(), typeof(GlobalStorageSection))) iglobal = c;
-                else
-                if (Tools.Reflection.IsTypeOf(child.GetType(), typeof(WorkingStorageSection))) iworking = c;
-                else
-                if (Tools.Reflection.IsTypeOf(child.GetType(), typeof(LocalStorageSection))) ilocal = c;
-                else
-                if (Tools.Reflection.IsTypeOf(child.GetType(), typeof(LinkageSection))) ilinkage = c;
-                c++;
+
+        private int UpdateSection(Node node)
+        {
+            int result = -1;
+            if (node is FileSection fileSection)
+            {
+                result = (int)SectionIndex.FileSection;
+                _sections[result] = fileSection;
             }
-            if (Tools.Reflection.IsTypeOf(section, typeof(GlobalStorageSection))) return Math.Max(0,ifile+1);
-            if (Tools.Reflection.IsTypeOf(section, typeof(WorkingStorageSection))) return Math.Max(0, Math.Max(ifile + 1, iglobal + 1));
-            if (Tools.Reflection.IsTypeOf(section, typeof(LocalStorageSection))) return Math.Max(0,Math.Max(Math.Max(ifile+1, iglobal+1), iworking + 1));
-            if (Tools.Reflection.IsTypeOf(section, typeof(LinkageSection))) return Math.Max(0, Math.Max(Math.Max(Math.Max(ifile + 1, iglobal + 1), iworking + 1), ilocal + 1));
-            return 0;
+            else if (node is GlobalStorageSection globalStorageSection)
+            {
+                result = (int)SectionIndex.GlobalStorageSection;
+                _sections[result] = globalStorageSection;
+            }
+            else if (node is WorkingStorageSection workingStorageSection)
+            {
+                result = (int)SectionIndex.WorkingStorageSection;
+                _sections[result] = workingStorageSection;
+            }
+            else if (node is LocalStorageSection localStorageSection)
+            {
+                result = (int)SectionIndex.LocalStorageSection;
+                _sections[result] = localStorageSection;
+            }
+            else if (node is LinkageSection linkageSection)
+            {
+                result = (int)SectionIndex.LinkageSection;
+                _sections[result] = linkageSection;
+            }
+
+            return result;
+        }
+
+        private int WhereShouldIAdd(int maxSectionIndex)
+        {
+            return _sections.Take(maxSectionIndex).Count(s => s != null);
```

Comment:
Could you please add a comment explaining what happens when `maxSectionIndex` is -1 ? I had to check the behavior of `Take` method when given count is inferior to 0

Reply:
Done